prow/tide/tide.go

```diff
@@ -1382,15 +1382,24 @@ func (c *syncController) nonFailedBatchForJobAndRefsExists(jobName string, refs
 }
 
 func (c *syncController) takeAction(sp subpool, batchPending, successes, pendings, missings, batchMerges []CodeReviewCommon, missingSerialTests map[int][]config.Presubmit) (Action, []CodeReviewCommon, error) {
+	updateMergeMetrics := func(merged []CodeReviewCommon) {
```

Comment:
can we use `sync.DoOnce` to guard against duplicated metrics update?

Reply:
We shouldn't ever have multiple updates with our current logic, but even if we later updated Tide to do multiple sets of merges per sync loop I don't think `sync.DoOnce` would be correct since this update is parameterized based on the number of PRs merged and multiple merges wouldn't be duplicates and should be counted.
In other words, if we did allow a serial merge and a batch merge to both occur I think we'd still want to update the metrics for each.

Is that the kind of duplicated update you are talking about guarding against or am I misunderstanding?

Reply:
what I referred to was a `defer`, as I'm more worries about someone updating this function and forgot to report metrics

Reply:
Gotcha, updated.